GitFyle.Core.Api/Models/Foundations/ContributionTypes/ContributionType.cs

```diff
@@ -0,0 +1,15 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+
+namespace GitFyle.Core.Api.Models.Foundations.ContributionTypes
+{
+    public class ContributionType
+    {
+        public Guid Id { get; set; }
+        public string Name { get; set; }
+        public int Value { get; set; }
+    }
```

Comment:
Add the navigation property to contributions
```cs
        public IEnumerable<Contribution> Contributions { get; set; }
```

Reply:
the navigation property 
` public virtual ICollection<Contribution> Contributions { get; set; }`
must be added in ContributionType Model?
as Microsoft docs mentions navigation property like so:
```
public class Course
{
  public int CourseID { get; set; }
  public string Title { get; set; }
  public int Credits { get; set; }
  public int DepartmentID { get; set; }
  public virtual Department Department { get; set; }
}

public class Department
{
   public Department()
   {
     this.Courses = new HashSet<Course>();
   }  
   public int DepartmentID { get; set; }
   public string Name { get; set; }
   public decimal Budget { get; set; }
   public DateTime StartDate { get; set; }
   public int? Administrator {get ; set; }
   public virtual ICollection<Course> Courses { get; set; }
}
```
Notice that it uses virtual and ICollection. What difference does it make though if I use ICollection instead of IEnummerable?

Reply:
See this [link](https://learn.microsoft.com/en-us/ef/core/modeling/relationships/navigations#collection-types)

Reply:
This is a very good question.   Either one can be used by EF.  It might be a nice discussion point for tonight.

`IEnumerable<T>` is a more basic interface and it is read-only. It does not provide methods to modify the collection, it only allows iteration over the collection.  This is perfect for us as we do not add or modify the collection. We use the foundation services that will handle the CRUD operations and validations for each entity.

In terms of EF the use of virtual will determine if you want lazy loading or eager loading.

**Use virtual:**
- When you want to enable lazy loading.
- When you need full change tracking capabilities via proxies.  (ICollection<T>)

**Do not use virtual:**
- When you are using eager loading (Include/ThenInclude).
- When you prefer explicit loading (Load method).
- When you want to avoid the overhead of proxy creation for performance reasons.

Reply:
> I cannot reference the contribution model as it doesn't exist in my code.

That is fine.  You can leave it out on this PR.  It will be merged before the the Contribution one and then we can add this as part of the navigation changes on Contribution